test/app/components/market/submit/submit-ncj-template.component.spec.ts

```diff
@@ -0,0 +1,215 @@
+import { Component, DebugElement, NO_ERRORS_SCHEMA } from "@angular/core";
```

Comment:
I started moving the test out of the test dir and putting them next to the file they are testing. I feel this makes test writing much easier and force us to have a better structure otherwise folder are overloaded with different files.

For a component I do a folder in this format:

component-name
   component-name.component.ts
   component-name.component.spec.ts
   component-name.html
   component-name.scss
   index.ts

I think we should change the test out of the test dir when we add a new one or update an existing one